lib/ansible/modules/remote_management/hpilo/hpilo_boot.py

```diff
@@ -0,0 +1,204 @@
+    default: admin
+  media:
+    description:
+    - The boot media to boot the system from
+    default: network
+    choices: [ "cdrom", "floppy", "hdd", "network", "normal", "usb" ]
+  image:
+    description:
+    - The URL of a cdrom, floppy or usb boot media image.
+      protocol://username:password@hostname:port/filename
+    - protocol is either 'http' or 'https'
+    - username:password is optional
+    - port is optional
+  state:
+    description:
+    - The state of the boot media.
+    - "no_boot: Do not boot from the device"
+    - "boot_once: Boot from the device once and then notthereafter"
+    - "boot_always: Boot from the device each time the serveris rebooted"
+    - "connect: Connect the virtual media device and set to boot_always"
+    - "disconnect: Disconnects the virtual media device and set to no_boot"
+    - "poweroff: Power off the server"
+    default: boot_once
+    choices: [ "boot_always", "boot_once", "connect", "disconnect", "no_boot", "poweroff" ]
+  force:
+    description:
+    - Whether to force a reboot (even when the system is already booted).
+    - As a safeguard, without force, hpilo_boot will refuse to reboot a server that is already running.
+    default: no
+    choices: [ "yes", "no" ]
+requirements:
+- hpilo
+notes:
+- To use a USB key image you need to specify floppy as boot media.
+- This module ought to be run from a system that can access the HP iLO
+  interface directly, either by using C(local_action) or using C(delegate_to).
+'''
+
+EXAMPLES = r'''
+- name: Task to boot a system using an ISO from an HP iLO interface only if the system is an HP server
+  hpilo_boot:
+    host: YOUR_ILO_ADDRESS
+    login: YOUR_ILO_LOGIN
+    password: YOUR_ILO_PASSWORD
+    media: cdrom
+    image: http://some-web-server/iso/boot.iso
+  when: cmdb_hwmodel.startswith('HP ')
+  delegate_to: localhost
+
+- name: Power off a server
+  hpilo_boot:
+    host: YOUR_ILO_HOST
+    login: YOUR_ILO_LOGIN
+    password: YOUR_ILO_PASSWORD
+    state: poweroff
+  delegate_to: localhost
+'''
+
+import sys
+import time
+import warnings
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    import hpilo
+    HAS_HPILO = True
+except ImportError:
+    HAS_HPILO = False
+
+
+# Suppress warnings from hpilo
+warnings.simplefilter('ignore')
+
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            host = dict(required=True, type='str'),
+            login = dict(default='Administrator', type='str'),
+            password = dict(default='admin', type='str', no_log=True),
+            media = dict(default=None, type='str', choices=['cdrom', 'floppy', 'rbsu', 'hdd', 'network', 'normal', 'usb']),
+            image = dict(default=None, type='str'),
+            state = dict(default='boot_once', type='str', choices=['boot_always', 'boot_once', 'connect', 'disconnect', 'no_boot', 'poweroff']),
+            force = dict(default='no', type='bool', choices=BOOLEANS),
```

Comment:
oh, that was a misunderstanding. I didn't want to change the default, I just wanted to suggest you choose the python boolean `False` instead of string 'no'. Sorry for the confusion.